### PR TITLE
[WIP] Pytorch Lightning support

### PR DESCRIPTION
**Pull request by sjrl**
I've started adding a Pytorch Lightning Connector. You can test it out in `padl/notebooks/03_padl_lightning.ipynb`.

Please leave any suggestions you may have on how to refactor or best the create the user interface for this.

### CODE REVIEW

**Comment by sjrl** [comment]
@wuhu The `test_padl_lightning` test is failing right now due to a saving error we talked about earlier.

**Comment by wuhu** [comment]
@sjrl bug is fixed

File: padl/transforms.py [hunk 1]
```diff
@@ -440,7 +440,7 @@ def _pd_find_varname(self, scopedict: dict) -> Optional[str]:
         except IndexError:
             return None
 
-    def pd_varname(self) -> Optional[str]:
+    def pd_varname(self, module=None) -> Optional[str]:
```
Comment (jasonkhadka) [inline padl/transforms.py:443]: description of module is missing.

**Comment by sjrl** [comment]
Will put functionality into separate repo

**Pull request closed without merging**